bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,872 @@
+
+                    ret = false;
+                } catch (Exception e1) {
+                    logger.debug("Error by opening Websocket connection [{}]", e1.getMessage());
+
+                    ret = false;
+                }
+            } else {
+                ret = false;
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to close the websocket connection
+     */
+    public void closeWebSocketConnection() {
+
+        socketMonitor.interrupt();
+
+        QueuedThreadPool localThreadPool = jettyThreadPool;
+
+        if (localThreadPool != null) {
+            try {
+                localThreadPool.stop();
+            } catch (Exception e1) {
+                logger.debug("Error by closing Websocket connection [{}]", e1.getMessage());
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient != null) {
+            try {
+                localWebSocketClient.stop();
+            } catch (Exception e2) {
+                logger.debug("Error by closing Websocket connection [{}]", e2.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Method to open the websocket connection
+     */
+    public boolean openWebSocketConnection() {
+        boolean ret = false;
+
+        QueuedThreadPool localThreadPool = jettyThreadPool;
+
+        if (localThreadPool == null) {
+            jettyThreadPool = new QueuedThreadPool();
+
+            localThreadPool = jettyThreadPool;
+
+            if (localThreadPool != null) {
+                localThreadPool.setName(FreeAtHomeBridgeHandler.class.getSimpleName());
+                localThreadPool.setDaemon(true);
+                localThreadPool.setStopTimeout(0);
+
+                ret = true;
+            }
+        }
+
+        WebSocketClient localWebSocketClient = websocketClient;
+
+        if (localWebSocketClient == null) {
+            websocketClient = new WebSocketClient(httpClient);
+
+            localWebSocketClient = websocketClient;
+
+            if (localWebSocketClient != null) {
+                localWebSocketClient.setExecutor(jettyThreadPool);
+
+                socketMonitor.start();
+
+                ret = true;
+            } else {
+                ret = false;
+            }
+        } else {
+            ret = true;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to initialize the bridge
+     *
+     *
+     */
+    @Override
+    public void initialize() {
+        scheduler.execute(() -> {
+            boolean thingReachable = true;
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            httpConnectionOK.set(false);
+
+            // load configuration
+            FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+            ipAddress = locConfig.ipAddress;
+            password = locConfig.password;
+            username = locConfig.username;
```

Comment:
The configuration should be validated. For these three parameters I would simply verify that they are not empty:
```suggestion
            ipAddress = locConfig.ipAddress;
            if (ipAddress.isBlank()) {
               updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                        "@text/conf-error.ip-address-missing");
                return;
            }
            password = locConfig.password;
            // See above
            username = locConfig.username;
            // See above
```

The validation should take place before setting status **UNKNOWN**.

Reply:
**Fixed** 

You have right it is sufficient to have the `initialize()`. Therefore removed